tests/unit/local/docker/test_lambda_build_container.py

```diff
@@ -0,0 +1,157 @@
+        container_dirs = get_container_dirs_mock.return_value = {
+            "source_dir": "/mysource",
+            "manifest_dir": "/mymanifest",
+            "artifacts_dir": "/myartifacts",
+            "scratch_dir": "/myscratch",
+        }
+
+        container = LambdaBuildContainer("protocol",
+                                         "language",
+                                         "dependency",
+                                         "application",
+                                         "/foo/source",
+                                         "/bar/manifest.txt",
+                                         "runtime",
+                                         optimizations="optimizations",
+                                         options="options",
+                                         log_level="log-level")
+
+        self.assertEquals(container.image, image)
+        self.assertEquals(container.executable_name, "lambda-builders")
+        self.assertEquals(container._entrypoint, entry)
+        self.assertEquals(container._cmd, [])
+        self.assertEquals(container._working_dir, container_dirs["source_dir"])
+        self.assertEquals(container._host_dir, "/foo/source")
+        self.assertEquals(container._env_vars, {"LAMBDA_BUILDERS_LOG_LEVEL": "log-level"})
+        self.assertEquals(container._additional_volumes, {
+            "/bar": {
+                "bind": container_dirs["manifest_dir"],
+                "mode": "ro"
+            }
+        })
+
+        self.assertEquals(container._exposed_ports, None)
+        self.assertEquals(container._memory_limit_mb, None)
+        self.assertEquals(container._network_id, None)
+        self.assertEquals(container._container_opts, None)
+
+        make_request_mock.assert_called_once()
+        get_entrypoint_mock.assert_called_once_with(request)
+        get_image_mock.assert_called_once_with("runtime")
+        get_container_dirs_mock.assert_called_once_with("/foo/source", "/bar")
+
+
+class TestLambdaBuildContainer_make_request(TestCase):
+
+    def test_must_make_request_object_string(self):
+
+        container_dirs = {
+            "source_dir": "source_dir",
+            "artifacts_dir": "artifacts_dir",
+            "scratch_dir": "scratch_dir",
+            "manifest_dir": "manifest_dir"
+        }
+
+        result = LambdaBuildContainer._make_request("protocol",
+                                                    "language",
+                                                    "dependency",
+                                                    "application",
+                                                    container_dirs,
+                                                    "manifest_file_name",
+                                                    "runtime",
+                                                    "optimizations",
+                                                    "options")
+
+        self.maxDiff = None  # Print whole json diff
```

Comment:
prolly dont need this?

Reply:
No, its helpful in debugging when the diff doesn't match up